diff -Naur a/steam b/steam
--- a/steam	2019-03-24 23:05:23.035074609 +0000
+++ b/steam	2019-03-24 23:15:11.768471262 +0000
@@ -17,6 +17,23 @@
 # Set up domain for script localization
 export TEXTDOMAIN=steam
 
+# Add paths to occasionally needed libraries not found in /usr/lib.
+export LD_LIBRARY_PATH+="${LD_LIBRARY_PATH+:}@@GENTOO_LD_LIBRARY_PATH@@"
+
+# Steam appends /usr/lib32 to LD_LIBRARY_PATH. We need to make sure that
+# OpenGL implementation dir goes before that, so we need to append it
+# to user's LD_LIBRARY_PATH ourselves. But that's needed only with
+# the new eselect-opengl that uses 000opengl file.
+if [[ -f /etc/env.d/000opengl ]]; then
+	. /etc/env.d/000opengl
+	# Append only when LDPATH is non-empty -- i.e. using nvidia or ati.
+	[[ -n ${LDPATH} ]] && LD_LIBRARY_PATH+=":${LDPATH}"
+fi
+
+# Preload Gentoo's execve wrapper so that steamwebhelper prefers system
+# libraries over older bundled libraries.
+[[ ${STEAM_RUNTIME} = 0 ]] && export LD_PRELOAD+="${LD_PRELOAD+:}@@GENTOO_LD_PRELOAD@@"
+
 function show_message()
 {
 	style=$1
