--- usr/bin/steam_orig	2012-12-14 16:51:53.429119063 +0100
+++ usr/bin/steam	2012-12-14 16:56:08.843216298 +0100
@@ -38,7 +38,7 @@
 		# Save the prompt in a temporary file because it can have newlines in it
 		tmpfile="$(mktemp || echo "/tmp/steam_message.txt")"
 		echo -e "$*" >"$tmpfile"
-		xterm -T "$title" -e "cat $tmpfile; echo -n 'Press enter to continue: '; read input"
+		${TERM} -T "$title" -e "cat $tmpfile; echo -n 'Press enter to continue: '; read input"
 		rm -f "$tmpfile"
 	fi
 }
@@ -178,7 +178,7 @@
 	umask "$omask"
 
 	# Try and install any extra packages
-	install_extra_packages jockey-common python-apt
+	#install_extra_packages jockey-common python-apt
 }
 
 function repair_bootstrap()
